### DISCUSSION TITLE
Ban package names starting with - or -- or —

### DISCUSSION THREAD

**Discussion by ThisIsMissEm** [npm/feedback#652] [Registry] [answered]
Currently folks can accidentally install packages that are named `-` by accident, which can [lead to security vulnerabilities](https://www.bleepingcomputer.com/news/software/empty-npm-package-has-over-700-000-downloads-heres-why/).

I'm suggesting that we should ban package names from starting with `-` or `--` (covered by just blocking `-`) and `—` as `--` in prose is often autocorrected to `—`.

This would help reduce attack surface where a mistyped command results in a malicious package being installed.

**Comment by MylesBorins** [comment, marked as answer]
Hey @ThisIsMissEm I've captured this internally and we are discussing next steps. I'll let you know if / when we prioritize this work and when you can expect it to land.

**Reply by ThisIsMissEm** [reply]
Nice one! I'm sure it'll be a good improvement, I might also suggest adding code to the npm cli to warn or just automatically remove the bad packages from package-lock.json and/or package.json
